src/components/@shared/FormInput/InputElement/ConsumerParameters/ConsumerParameterInput.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { ReactElement } from 'react'
+import Input, { InputProps } from '../..'
+import { Field, useField, useFormikContext } from 'formik'
+import {
+  FormConsumerParameter,
+  FormPublishData
+} from '@components/Publish/_types'
+import { defaultConsumerParam } from '.'
+
+export default function ConsumerParameterInput({
+  index,
+  inputName,
+  ...props
+}: InputProps & {
+  index: number
+  inputName: string
+}): ReactElement {
+  const { setFieldTouched } = useFormikContext<FormPublishData>()
+  const [field, meta, helpers] = useField<FormConsumerParameter[]>(inputName)
+
+  const resetDefaultValue = (
+    parameterName: string,
+    parameterType: FormConsumerParameter['type'],
+    index: number
+  ) => {
+    if (parameterName !== 'type') return
+    if (field.value[index].type === parameterType) return
+
+    setFieldTouched(`${field.name}[${index}].default`, false)
+    helpers.setValue(
+      field.value.map((currentParam, i) => {
+        if (i !== index) return currentParam
+
+        return {
+          ...defaultConsumerParam,
+          ...currentParam,
+          default: defaultConsumerParam.default
+        }
+      })
+    )
+  }
+
+  const getFieldValue = (
+    fieldName: string,
+    parameter: FormConsumerParameter
+  ) => {
+    // split the input name to every "." and keep the last
+    // word which corresponds to the field name
+    const valueKey = fieldName.split('.').slice(-1)[0]
+    const value = parameter[valueKey]
+
+    // convert the boolean values assigned to the "required" field
+    // into the select component options ["required", "optional"]
+    if (valueKey !== 'required' || typeof value === 'string') return value
+
+    return value ? 'required' : 'optional'
+  }
```

Comment:
thought: another possible split here is the `RequiredInput` as this functionality is basically only needed for the `required` param property.